pandas/_libs/interval.pyx

```diff
@@ -1,5 +1,6 @@
 import numbers
 from operator import le, lt
+from datetime import timedelta
```

Comment:
Cython may not like this import judging from the failing CI, not sure why though

Reply:
cython should be OK with this, but we actually want to do 

```
from cpython.datetime cimport PyDelta_Check, PyDateTime_IMPORT
PyDateTime_IMPORT
```

then below the `isinstance(obj, (Timedelta, timedelta))` becomes `PyDelta_Check(obj)`

Reply:
@jbrockmendel I had to remove the np.timedelta64 + pd.Interval test case because it was seemingly raising in numpy before ever getting to pandas (I could open a follow-up issue, although I'm not sure if there's anything we can do about that; the other direction pd.Interval + np.timedelta64 works fine)

Reply:
> the np.timedelta64 + pd.Interval test case because it was seemingly raising in numpy before ever getting to pandas (

can you show the traceback?

Reply:
```python
[ins] In [3]: interval                                                                                                                                                                                       
Out[3]: Interval('2017-01-01', '2018-01-01', closed='right')

[ins] In [4]: delta                                                                                                                                                                                          
Out[4]: numpy.timedelta64(7,'D')

[ins] In [5]: interval + delta                                                                                                                                                                               
Out[5]: Interval('2017-01-08', '2018-01-08', closed='right')

[ins] In [6]: delta + interval                                                                                                                                                                               
---------------------------------------------------------------------------
UFuncTypeError                            Traceback (most recent call last)
<ipython-input-6-2d6b9ea7a88b> in <module>
----> 1 delta + interval

UFuncTypeError: ufunc 'add' cannot use operands with types dtype('<m8[D]') and dtype('O')
```
Seems to come from here: https://github.com/numpy/numpy/blob/master/numpy/core/_exceptions.py#L41

Reply:
i think this will be resolved by defining `Interval.__array_priority__ = 1000`

Reply:
Works, thanks!